homeassistant/components/device_tracker/__init__.py

```diff
@@ -143,22 +143,33 @@ def async_setup_platform(p_type, p_config, disc_info=None):
             return
 
         try:
-            if hasattr(platform, 'get_scanner'):
+            scanner = None
+            setup = None
+            if hasattr(platform, 'async_get_scanner'):
+                scanner = yield from platform.async_get_scanner(
+                    hass, {DOMAIN: p_config})
+            elif hasattr(platform, 'get_scanner'):
                 scanner = yield from hass.loop.run_in_executor(
                     None, platform.get_scanner, hass, {DOMAIN: p_config})
+            elif hasattr(platform, 'async_setup_scanner'):
+                setup = yield from platform.setup_scanner(
+                    hass, p_config, tracker.see)
+            elif hasattr(platform, 'setup_scanner'):
+                setup = yield from hass.loop.run_in_executor(
+                    None, platform.setup_scanner, hass, p_config, tracker.see)
+            else:
+                raise HomeAssistantError("Invalid device_tracker platform.")
 
-                if scanner is None:
-                    _LOGGER.error('Error setting up platform %s', p_type)
-                    return
+            if scanner is None and not setup:
```

Comment:
Move this test to below the `if scanner:` test below and it can simply become `if not setup:`.